wellcomecollection/storage-service
# Migrate Ingest Inspector to AWS

## What does this change?

This migrates the Wellcome Ingest Inspector from Glitch into AWS. As part of this migration, the following changes were made:

1) The application and infrastructure code were moved from their [original repository](https://github.com/wellcomecollection/ingest-inspector) into the storage-service repository
2) The backend was migrated from Flask to a single Lambda function (placed behind an API Gateway endpoint)
3) The frontend was migrated from Flask templates to a static Next.js app (hosted from an S3 bucket via CloudFront)
4) Visual changes were made in the frontend, borrowing visual components from wellcomecollection.org, to make the app more consistent with Wellcome Collection's design system


## How to test

Visit the deployed version on https://d1dti1kbjcy26.cloudfront.net and verify that it provides the same functionality as [the original version](https://wellcome-ingest-inspector.glitch.me).

## How can we measure success?

The Vercel application has been decommissioned and it's credentials removed. Developers & digital production users are both able to use this new application

## Have we considered potential risks?

This application is public facing, but as discussed in the README for this project that risk is mitigated by the lack of secret information in the project. A future improvement would be to put this behind a staff login.
